src/template.rs

```diff
@@ -91,8 +102,59 @@ impl Template {
         self
     }
 
-    pub fn render(self) -> String {
-        let (year_repl, author_repl, ident_repl) = if self.spdx_template {
+    pub fn outdated_license_pattern(&self, commenter: &Box<dyn Comment>, columns: Option<usize>) -> Regex {
+        self.build_year_varying_regex(&mut self.content.clone(), commenter, columns, false)
+    }
+
+    pub fn outdated_license_trimmed_pattern(&self, commenter: &Box<dyn Comment>, columns: Option<usize>) -> Regex {
+        self.build_year_varying_regex(&mut self.content.clone(), commenter, columns, true)
+    }
+
+    pub fn render(&self) -> String {
+        let (year_repl, author_repl, ident_repl) = self.replacement_tokens();
+        let template = self.unwrap_license_headers(&mut self.content.clone());
+
+        // Perform our substitutions
+        template
+            .replace(year_repl, &self.context.get_year())
+            .replace(author_repl, &self.context.get_authors())
+            .replace(ident_repl, &self.context.ident)
+    }
+
+    fn build_year_varying_regex(&self, content: &mut str, commenter: &Box<dyn Comment>, columns: Option<usize>, trim_trailing: bool) -> Regex {
+        let (year_repl, author_repl, ident_repl) = self.replacement_tokens();
+        let mut rendered = commenter.comment(
```

Comment:
It feels to me like there's an opportunity to refactor the render method so we can reduce the duplicate code here. Make it rely less on `self.context` and pass in the `context` with a public render that just passes in `self.context` as an example. Then you just need to `clone` context and pass in the `INTERMEDIATE_YEAR_TOKEN` and can reduce some steps.

Reply:
I moved the common bits into a functioned called `interpolate()` and reused that in both `render()` and `build_year_varying_regex()`.

I chose to do that this way because modifying `render()` to take a `Context` would have forced me to make `Template.context` public, and I erred on the side of not exposing that internal state unnecessarily, and this accomplishes the same goal to share implementation/DRY-up while maintaining visibility boundaries.